pkg/apis/core/validation/events.go

```diff
@@ -33,8 +37,77 @@ const (
 	NoteLengthLimit              = 1024
 )
 
-// ValidateEvent makes sure that the event makes sense.
-func ValidateEvent(event *core.Event) field.ErrorList {
+func ValidateEventCreate(event *core.Event, requestVersion schema.GroupVersion) field.ErrorList {
+	// Make sure events always pass legacy validation.
+	allErrs := legacyValidateEvent(event)
+	if requestVersion == core.SchemeGroupVersion || requestVersion == eventsv1beta1.SchemeGroupVersion {
+		// No further validation for backwards compatibility.
+		return allErrs
+	}
+
+	// Strict validation applies to creation via events.k8s.io/v1 API and newer.
+	allErrs = append(allErrs, ValidateObjectMeta(&event.ObjectMeta, true, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, validateV1EventSpec(event)...)
+	return allErrs
+}
+
+func ValidateEventUpdate(newEvent, oldEvent *core.Event, requestVersion schema.GroupVersion) field.ErrorList {
+	// Make sure the new event always passes legacy validation.
+	allErrs := legacyValidateEvent(newEvent)
+	if requestVersion == core.SchemeGroupVersion || requestVersion == eventsv1beta1.SchemeGroupVersion {
+		// No further validation for backwards compatibility.
+		return allErrs
+	}
+
+	// Strict validation applies to update via events.k8s.io/v1 API and newer.
+	allErrs = append(allErrs, ValidateObjectMetaUpdate(&newEvent.ObjectMeta, &oldEvent.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, validateV1EventSpec(newEvent)...)
```

Comment:
calling `validateV1EventSpec` here means an event that was created via `events.k8s.io/v1beta1` or `core/v1`, and does not pass strict `events.k8s.io/v1` validation, cannot be updated via the `events.k8s.io/v1` API.

Since we're already ensuring almost all the fields are not permitted to change in update requests, would be be sufficient to only validate the fields which are allowed to change (series.count, series.lastObservedTime) and to tolerate update requests that leave those field values unchanged?

Reply:
Make sense. I only added series validation as a common function and moved all others validations to ValidateEventCreate.